src/integrations/GoogleAds/browser.js

```diff
@@ -70,45 +78,41 @@ class GoogleAds {
   track(rudderElement) {
     logger.debug('in GoogleAdsAnalyticsManager track');
 
-    // Dynamic remarketing disabled
-    if (!this.dynamicRemarketing) {
-      const conversionData = this.getConversionData(
-        this.clickEventConversions,
-        rudderElement.message.event,
-      );
-      if (conversionData.conversionLabel) {
-        const { conversionLabel } = conversionData;
-        const { eventName } = conversionData;
-        const sendToValue = `${this.conversionId}/${conversionLabel}`;
-        let properties = {};
-        if (rudderElement.message.properties) {
-          properties.value = rudderElement.message.properties.revenue;
-          properties.currency = rudderElement.message.properties.currency;
-          properties.transaction_id = rudderElement.message.properties.order_id;
-        }
-        properties.send_to = sendToValue;
-        properties = removeUndefinedAndNullValues(properties);
-        window.gtag('event', eventName, properties);
-      }
-    } else {
-      const { event } = rudderElement.message;
-      if (!event) {
-        logger.error('Event name not present');
-        return;
+    const { event } = rudderElement.message;
+    const conversionData = this.getConversionData(this.clickEventConversions, event);
+    if (conversionData.conversionLabel && this.sendEvent('conversion', event)) {
+      const { conversionLabel } = conversionData;
+      const { eventName } = conversionData;
+      const sendToValue = `${this.conversionId}/${conversionLabel}`;
+      let properties = {};
+      if (rudderElement.message.properties) {
+        properties.value = rudderElement.message.properties.revenue;
+        properties.currency = rudderElement.message.properties.currency;
+        properties.transaction_id = rudderElement.message.properties.order_id;
       }
+      properties.send_to = sendToValue;
+      properties = removeUndefinedAndNullValues(properties);
+      window.gtag('event', eventName, properties);
+    }
+
+    if (!event) {
+      logger.error('Event name not present');
+      return;
+    }
```

Comment:
Shouldn't this be checked before  L82?

Reply:
for conversion events if event name is not there then we are taking default event name which is `Viewed a Page` from `getConversionData` but for this also `defaultPageConversion` label is required from config. that's why we haven't added condition there

Reply:
Yeah but can we add some context (js docs) on the `getConversionData` for this?

Reply:
you mean add comments that if event name is not there then we are taking default name from `getConversionData` function?

Reply:
Yes add docs to function as intent is not pretty clear on it